src/mongoc/mongoc-topology.c

```diff
@@ -321,8 +349,10 @@ mongoc_topology_select (mongoc_topology_t         *topology,
                                                            error);
 
       if (! selected_server) {
-         /* TODO request an immediate topology check here */
-         r = mongoc_cond_timedwait (&topology->cond_client, &topology->mutex, expire_at - now);
+         topology->scan_requested = true;
```

Comment:
we shouldn't have to do this.   I just got hung up on the fact that we have the mutex here.  It's fine to either have a version of request_scan which assumes we have the lock, or to drop the lock and use request scan, then resume it